### PR TITLE
[EOS-10845] Feature/eos 10847 cluster autoescaler integration

### PR DESCRIPTION
**Pull request by lreciomelero**

### CODE REVIEW

File: pkg/cluster/internal/create/actions/cluster/templates/aws.eks.tmpl [hunk 1]
```diff
@@ -70,6 +70,13 @@ spec:
 apiVersion: cluster.x-k8s.io/v1beta1
 kind: MachineDeployment
 metadata:
+  annotations:
+  {{- if ne $node.NodeGroupMaxSize "" }} 
+    cluster.x-k8s.io/cluster-api-autoscaler-node-group-max-size: '{{ $node.NodeGroupMaxSize }}'  
+  {{- end }}
+  {{- if ne $node.NodeGroupMinSize "" }} 
+    cluster.x-k8s.io/cluster-api-autoscaler-node-group-min-size: '{{ $node.NodeGroupMinSize }}' 
+  {{- end }}
```
Comment (stg-0) [inline pkg/cluster/internal/create/actions/cluster/templates/aws.eks.tmpl:79]: max and min values should be divided by 3 (number of zones), perhaps with a new function..

File: pkg/cluster/internal/create/actions/createworker/utils.go [hunk 2]
```diff
@@ -196,3 +198,15 @@ func deleteKey(key string, descriptorMap map[string]interface{}) {
 		delete(descriptorMap, key)
 	}
 }
+
+func integrateClusterAutoscaler(node nodes.Node, kubeconfigPath string, clusterID string, provider string) exec.Cmd {
+	cmd := node.Command("helm", "install", "autoescaler-release", "autoscaler/cluster-autoscaler",
```
Comment (stg-0) [inline pkg/cluster/internal/create/actions/createworker/utils.go:203]:
```suggestion
	cmd := node.Command("helm", "install", "cluster-autoscaler", "autoscaler/cluster-autoscaler",
```

File: pkg/cluster/internal/create/actions/createworker/createworker.go [hunk 3]
```diff
@@ -293,13 +310,9 @@ spec:
...
 	cmd = node.Command("sh", "-c", "clusterctl move -n "+capiClustersNamespace+" --to-kubeconfig "+kubeconfigPath)
-	cmd.SetEnv("AWS_REGION="+aws.Credentials.Region,
-		"AWS_ACCESS_KEY_ID="+aws.Credentials.AccessKey,
-		"AWS_SECRET_ACCESS_KEY="+aws.Credentials.SecretKey,
-		"AWS_B64ENCODED_CREDENTIALS="+generateB64Credentials(aws.Credentials.AccessKey, aws.Credentials.SecretKey, aws.Credentials.Region),
-		"GITHUB_TOKEN="+github_token)
+
 	if err := cmd.SetStdout(&raw).Run(); err != nil {
-		return errors.Wrap(err, "failed to pivot management role to worker cluster")
+		return errors.Wrap(err, "failed to transfer cluster")
```
Comment (stg-0) [inline pkg/cluster/internal/create/actions/createworker/createworker.go:315]: AFAIK, the operation in CAPI is called "pivoting".

File: pkg/cluster/internal/providers/docker/stratio/Dockerfile [hunk 4]
```diff
@@ -17,3 +18,7 @@ RUN curl -L https://github.com/kubernetes-sigs/cluster-api-provider-aws/releases
 
 # Add alias for kubectl
 RUN echo 'alias k="kubectl"' >> ~/.bashrc
+
+RUN curl -L https://get.helm.sh/${HELM_FILENAME} -o /root/helm.tar.gz && tar -xf /root/helm.tar.gz -C /root && mv /root/linux-amd64/helm /usr/local/bin/helm && rm -rf /root/linux-amd64 && rm /root/helm.tar.gz \
```
Comment (stg-0) [inline pkg/cluster/internal/providers/docker/stratio/Dockerfile:22]:
```suggestion
RUN curl -L https://get.helm.sh/helm-${HELM_VERSION}-linux-amd64.tar.gz -o /root/helm.tar.gz && tar -xf /root/helm.tar.gz -C /root && mv /root/linux-amd64/helm /usr/local/bin/helm && rm -rf /root/linux-amd64 && rm /root/helm.tar.gz \
```

File: pkg/cluster/internal/providers/docker/stratio/Dockerfile [hunk 5]
```diff
@@ -3,7 +3,8 @@ FROM kindest/node:v1.24.6
 # Tools versions
 ENV CLUSTERCTL=v1.3.2
 ENV CLUSTERAWSADM=v2.0.2
-
+ENV HELM_VERSION=v3.11.0
+ENV HELM_FILENAME=helm-${HELM_VERSION}-linux-amd64.tar.gz
 # Init feature gates
```
Comment (stg-0) [inline pkg/cluster/internal/providers/docker/stratio/Dockerfile:8]:
```suggestion
ENV HELM_VERSION=v3.11.0

# Init feature gates
```

**Review by stg-0 — CHANGES_REQUESTED** [review]

File: pkg/cluster/internal/create/actions/cluster/cluster.go [hunk 6]
```diff
@@ -106,6 +106,8 @@ type WorkerNodes []struct {
 	AZ               string `yaml:"az"`
 	SSHKey           string `yaml:"ssh_key"`
 	Spot             bool   `yaml:"spot" validate:"boolean"`
+	NodeGroupMaxSize string `yaml:"node_max_size"`
+	NodeGroupMinSize string `yaml:"node_min_size"`
```
[the remark this answers is not in the document]
Comment (lreciomelero) [reply pkg/cluster/internal/create/actions/cluster/cluster.go:110]: OK

File: pkg/cluster/internal/create/actions/cluster/templates/aws.eks.tmpl (hunk 1 above)
Comment (lreciomelero) [reply pkg/cluster/internal/create/actions/cluster/templates/aws.eks.tmpl:79]: OK

File: pkg/cluster/internal/create/actions/createworker/createworker.go (hunk 3 above)
Comment (lreciomelero) [reply pkg/cluster/internal/create/actions/createworker/createworker.go:315]: OK

File: pkg/cluster/internal/create/actions/createworker/utils.go (hunk 2 above)
Comment (lreciomelero) [reply pkg/cluster/internal/create/actions/createworker/utils.go:203]: OK

File: pkg/cluster/internal/providers/docker/stratio/Dockerfile (hunk 4 above)
Comment (lreciomelero) [reply pkg/cluster/internal/providers/docker/stratio/Dockerfile:22]: OK

File: pkg/cluster/internal/create/actions/createworker/createworker.go [hunk 7]
```diff
@@ -280,9 +279,27 @@ spec:
 
 	ctx.Status.End(true) // End Installing CAPx in worker cluster
 
+	ctx.Status.Start("Adding Cluster-Autoescaler 🗝️")
```
Comment (stg-0) [inline pkg/cluster/internal/create/actions/createworker/createworker.go:282]:
```suggestion
	ctx.Status.Start("Adding Cluster-Autoescaler 🗚")
```

**Review by stg-0 — APPROVED** [review]
LGTM

**Pull request merged**